src/components/Text.tsx

```diff
@@ -0,0 +1,49 @@
+import styled, { css } from "styled-components/macro";
+import { ThemeColors } from "../styles/theme";
+
+interface TextProps {
+  color?: keyof ThemeColors;
```

Comment:
Is it possible for this to be `keyof ThemeColors` or `string`? So that it supports looking up the theme colors (`color="neutral"`) as well as manually overriding with hex colors (`color="#ff0000"`)?

Reply:
so i _think_ you could do `color?: keyof ThemeColors | string;`

but there might be an issue using union types (`|`) with object keys:
https://github.com/microsoft/TypeScript/issues/12948

Reply:
hmm, yeah, I will need some more help with this, because that is what I tried first and got the following error that suggests the type should actually be `any`? 

```sh
Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{ nearBlack: string; darkNeutral: string; neutral: string; lightNeutral: string; lighterNeutral: string; white: string; mint: string; yellow: string; }'.
  No index signature with a parameter of type 'string' was found on type '{ nearBlack: string; darkNeutral: string; neutral: string; lightNeutral: string; lighterNeutral: string; white: string; mint: string; yellow: string; }'.
```

Reply:
let's pair on this sometime! i HATE typescript's errors. I think the important part here is
`expression of type 'string' can't be used to index type '{ nearBlack: ...`

Essentially this is a known object shape with specific keys, like `"nearBlack"` and `"darkNeutral"`. TS can't use any old string, like `"fishWithTeethLikeHumans"`, as a key on it. Only keys one the object can be used.

Reply:
ooo i have an example of what you might want to do! `keyof`!

e.g. here's the `keyof`: https://github.com/chrisbolin/perf-land/blob/master/src/state.ts#L59
and here's it in use: https://github.com/chrisbolin/perf-land/blob/master/src/state.ts#L206

Reply:
oh duh you are already doing this. so yeah let's pair, because now you are doing something more interesting